src/main/java/morethanhidden/restrictedportals/RestrictedPortals.java

```diff
@@ -61,20 +67,21 @@ public void preInit(FMLPreInitializationEvent event) {
 				endItem = GameRegistry.findItem(endSplit[0], endSplit[1]);
 				netherItem = GameRegistry.findItem(netherSplit[0], netherSplit[1]);
 			}else{
-
-				GameRegistry.registerItem(endKey, "endKey");
-				GameRegistry.registerItem(netherKey, "netherKey");
+				if (endLock)
+					GameRegistry.registerItem(endKey, "endKey");
+				if (netherLock)
+					GameRegistry.registerItem(netherKey, "netherKey");
 
 				endItem = RestrictedPortals.endKey;
 				netherItem = RestrictedPortals.netherKey;
 			}
 
 			//If Configuration is invalid
-			if (endItem == null){
+			if (endLock && endItem == null){
```

Comment:
null check only valid if both are null, would need to separate the lock check and give a default value.